src/test/java/net/sf/jsqlparser/statement/truncate/TruncateTest.java

```diff
@@ -41,6 +41,23 @@ public void testTruncate() throws Exception {
         assertEquals(statement, truncate.toString());
     }
 
+    @Test
+    public void testTruncatePostgresqlWithoutTableName() throws Exception {
```

Comment:
JSqlParser tries to rebuild the original SQL from its object tree. With your change, there would be always a `table` part. Use `assertSqlCanBeParsedAndDeparsed(statement)` to ensure the rebuilding behaviour. You have to introduce a flag, if the `table` keyword is used or not and then use this in the `toString` method of `Truncate` and the fitting deparser.

Reply:
I see. I updated the PR with the flags (also for `ONLY`). 

btw.
I also noticed there is a bit of an overlap between the `StatementDeparser` `visit(Truncate t)` and the `toString()` of the `Truncate`.  I think you want to keep things loosely coupled, but added a `deparse()` method to the `Statement` interface might be a way to prevent this code duplicating, as the toString() could simply call `deparse()`. The reason I would be tempted do this, is that the project _forces_ you to have a deparse implementation for each statement.